### PR TITLE
Added e2e tests for server and bidi stream cancellation.

### PR DESCRIPTION
**Pull request by a11r**

### CODE REVIEW

**Comment by a11r** [comment]
Note that these tests pass just fine, so it should be OK to merge. The failing client streaming test is in a separate PR.

**Comment by a11r** [comment]
This test is flaky due to the serve's OK status racing with the cancellation attempt at the client. No underlying bug here. I will fix the test in a few minutes. Please do not merge.

File: test/cpp/end2end/end2end_test.cc [hunk 1]
```diff
@@ -491,6 +491,72 @@ TEST_F(End2endTest, ServerCancelsRpc) {
   EXPECT_TRUE(s.details().empty());
 }
 
+// Client cancels server stream after sending some messages
+  TEST_F(End2endTest, ClientCancelsResponseStream) {
```
Comment (ctiller) [inline test/cpp/end2end/end2end_test.cc]: Indentation

**Comment by ctiller** [comment]
LGTM once formatting is fixed

**Comment by ctiller** [comment]
pinging on the formatting: this would be good to get in

**Comment by a11r** [comment]
Fixed. PTAL

**Pull request merged**